src/app/modules/ewt-patron/enrolment/not-enroled/not-enroled.component.ts

```diff
@@ -1,31 +1,55 @@
-import { Component } from '@angular/core';
-import { FormControl, Validators } from '@angular/forms';
-import { Store } from '@ngrx/store';
-import { RoleEnrolmentActions } from '@state';
-import { MatDialog } from '@angular/material/dialog';
-import { ConfirmationDialogComponent } from '../confirmation-dialog/confirmation-dialog.component';
-import { truthy } from '@operators';
+import { Component } from "@angular/core";
+import {
+  AbstractControl,
+  FormControl,
+  ValidationErrors,
+  ValidatorFn,
+  Validators,
+} from "@angular/forms";
+import { Store } from "@ngrx/store";
+import { RoleEnrolmentActions } from "@state";
+import domains from "disposable-email-domains";
 
-const emailPattern = '(?:[a-z0-9!#$%&\'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&\'*+/=?^_`{|}~-]+)*|"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])';
+const emailPattern =
+  "(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])";
+const EMAIL_DOMAINS_WHITELIST = "yopmail.com;yopmail.fr";
 
 @Component({
-  selector: 'app-not-enroled',
-  templateUrl: './not-enroled.component.html',
-  styleUrls: ['./not-enroled.component.scss']
+  selector: "app-not-enroled",
+  templateUrl: "./not-enroled.component.html",
+  styleUrls: ["./not-enroled.component.scss"],
 })
 export class NotEnroledComponent {
   // TODO: validate against the same email domain blacklist used on backend against disposable emails
-  email = new FormControl('', [Validators.pattern(emailPattern), Validators.required]);
+  email = new FormControl("", [
```

Comment:
I see you are using different code styling.

Reply:
Yeah, when it formats it automatically..it's visual code who does this.

Reply:
Prettier with eslint

Reply:
Ok, i'm using webstorm with defaults. Let's go with it as it is. Maybe in Switchboard we will integrate eslint + prettier